craft_application/services/service_factory.py

```diff
@@ -15,14 +15,41 @@
 from __future__ import annotations
 
 import dataclasses
+import importlib
+import re
 import warnings
-from typing import TYPE_CHECKING, Any
+from typing import (
+    TYPE_CHECKING,
+    Annotated,
+    Any,
+    ClassVar,
+    Literal,
+    TypeVar,
+    cast,
+    overload,
+)
+
+import annotated_types
```

Comment:
If you import this directly, shouldn't it be added to `pyproject.toml` as a dependency?

Reply:
Good call!